packages/webcert/src/feature/careProvider/CareProviderModal.tsx

```diff
@@ -1,23 +1,53 @@
-import { PopUpModal, ResourceLinkType } from '@frontend/common'
+import { CustomButton, ModalBase, ResourceLinkType } from '@frontend/common'
 import _ from 'lodash'
-import React from 'react'
-import { useSelector } from 'react-redux'
-import { getUserResourceLinks, selectIsLoadingUserStatistics } from '../../store/user/userSelectors'
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { updateIsCareProviderModalOpen } from '../../store/user/userActions'
+import { selectIsLoadingUserStatistics, getIsCareProviderModalOpen, getUserResourceLinks } from '../../store/user/userSelectors'
 import { CareProviderModalContent } from './CareProviderModalContent'
 
 const CareProviderModal: React.FC = () => {
+  const dispatch = useDispatch()
   const isLoadingUserStatistics = useSelector(selectIsLoadingUserStatistics, _.isEqual)
+  const isCareProviderModalOpen = useSelector(getIsCareProviderModalOpen)
   const userLinks = useSelector(getUserResourceLinks)
+
   const chooseUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHOOSE_UNIT)
+  const changeUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHANGE_UNIT)
+
+  const getModalTitle = () => {
+    if (chooseUnitLink) {
```

Comment:
It seems a bit weird that the purpose (title) of the modal is determined by the presence of resource links. Am I to understand that once a unit is chosen `ResourceLinkType.CHOOSE_UNIT` is no longer present?

Reply:
Correct, when the user have chosen a unit or if a unit is already set, ResourceLinkType.CHOOSE_UNIT is no longer present.